pyomo/contrib/solver/base.py

```diff
@@ -464,7 +502,15 @@ def _solution_handler(
         """Method to handle the preferred action for the solution"""
         symbol_map = SymbolMap()
         symbol_map.default_labeler = NumericLabeler('x')
-        model.solutions.add_symbol_map(symbol_map)
+        try:
+            model.solutions.add_symbol_map(symbol_map)
+        except AttributeError:
+            # Something wacky happens in IDAES due to the usage of ScalarBlock
+            # instead of PyomoModel. This is an attempt to fix that.
+            from pyomo.core.base.PyomoModel import ModelSolutions
+
+            setattr(model, 'solutions', ModelSolutions(model))
+            model.solutions.add_symbol_map(symbol_map)
```

Comment:
This seems weird to include here. Is this pointing to an edge case that was missed in #110 that should be fixed directly in the `Block` or `_BlockData` class?

Reply:
Absolutely, yes, it should be. It was something that Andrew ran into while trying to get this working for IDAES, since they inherit from `ScalarBlock`

Reply:
This is really wacky - and probably *not* something that should be added to Block (in the new world order, there should never be a reason to attach a symbolmap to the model)